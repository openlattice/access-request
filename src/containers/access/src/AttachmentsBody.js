// @flow
import { useState } from 'react';

import { Tab, Tabs } from 'lattice-ui-kit';

import Dropzone from '../../../components/Dropzone';
import ModalBody from '../../../components/styled';
import TabPanel from '../../../components/TabPanel';

const AttachmentsBody = () => {
  const [value, setValue] = useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  return (
    <ModalBody>
      <Tabs
          onChange={handleChange}
          value={value}
          indicatorColor="primary"
          textColor="primary">
        <Tab label="Manage" />
        <Tab label="Upload" />
      </Tabs>
      <TabPanel
          value={value}
          index={0}>
        manage
      </TabPanel>
      <TabPanel
          value={value}
          index={1}>
        <Dropzone onDrop={() => {}} />
      </TabPanel>
    </ModalBody>
  );
};

export default AttachmentsBody;
