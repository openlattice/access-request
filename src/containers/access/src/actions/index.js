/*
 * @flow
 */

import { newRequestSequence } from 'redux-reqseq';
import type { Map } from 'immutable';
import type { RequestSequence } from 'redux-reqseq';

const GET_FORMS :'GET_FORMS' = 'GET_FORMS';
const getForms :RequestSequence = newRequestSequence(GET_FORMS);

const GET_ALL_ACCESS_REQUESTS :'GET_ALL_ACCESS_REQUESTS' = 'GET_ALL_ACCESS_REQUESTS';
const getAllAccessRequests :RequestSequence = newRequestSequence(GET_ALL_ACCESS_REQUESTS);

const GET_ACCESS_REQUEST :'GET_ACCESS_REQUEST' = 'GET_ACCESS_REQUEST';
const getAccessRequest :RequestSequence = newRequestSequence(GET_ACCESS_REQUEST);

const SUBMIT_ACCESS_REQUEST :'SUBMIT_ACCESS_REQUEST' = 'SUBMIT_ACCESS_REQUEST';
const submitAccessRequest :RequestSequence = newRequestSequence(SUBMIT_ACCESS_REQUEST);

const SELECT_ACCESS_REQUEST :'SELECT_ACCESS_REQUEST' = 'SELECT_ACCESS_REQUEST';
const selectAccessRequest = (value :Map) => ({
  type: SELECT_ACCESS_REQUEST,
  value
});

const CLEAR_ACCESS_REQUEST :'CLEAR_ACCESS_REQUEST' = 'CLEAR_ACCESS_REQUEST';
const clearAccessRequest = () => ({
  type: CLEAR_ACCESS_REQUEST
});

const UPDATE_ACCESS_REQUEST :'UPDATE_ACCESS_REQUEST' = 'UPDATE_ACCESS_REQUEST';
const updateAccessRequest :RequestSequence = newRequestSequence(UPDATE_ACCESS_REQUEST);

const GET_ATTACHMENTS :'GET_ATTACHMENTS' = 'GET_ATTACHMENTS';
const getAttachments :RequestSequence = newRequestSequence(GET_ATTACHMENTS);

const UPLOAD_ATTACHMENTS :'UPLOAD_ATTACHMENTS' = 'UPLOAD_ATTACHMENTS';
const uploadAttachments :RequestSequence = newRequestSequence(UPLOAD_ATTACHMENTS);

const DELETE_ATTACHMENTS :'DELETE_ATTACHMENTS' = 'DELETE_ATTACHMENTS';
const deleteAttachments :RequestSequence = newRequestSequence(DELETE_ATTACHMENTS);

const SAVE_NEW_FORM_TEMPLATE :'SAVE_NEW_FORM_TEMPLATE' = 'SAVE_NEW_FORM_TEMPLATE';
const saveNewFormTemplate :RequestSequence = newRequestSequence(SAVE_NEW_FORM_TEMPLATE);

const UPDATE_ATTACHMENT_TAG :'UPDATE_ATTACHMENT_TAG' = 'UPDATE_ATTACHMENT_TAG';
const updateAttachmentTag :RequestSequence = newRequestSequence(UPDATE_ATTACHMENT_TAG);

export {
  CLEAR_ACCESS_REQUEST,
  DELETE_ATTACHMENTS,
  GET_ACCESS_REQUEST,
  GET_ALL_ACCESS_REQUESTS,
  GET_ATTACHMENTS,
  GET_FORMS,
  SAVE_NEW_FORM_TEMPLATE,
  SELECT_ACCESS_REQUEST,
  SUBMIT_ACCESS_REQUEST,
  UPDATE_ACCESS_REQUEST,
  UPDATE_ATTACHMENT_TAG,
  UPLOAD_ATTACHMENTS,
  clearAccessRequest,
  deleteAttachments,
  getAccessRequest,
  getAllAccessRequests,
  getAttachments,
  getForms,
  saveNewFormTemplate,
  selectAccessRequest,
  submitAccessRequest,
  updateAccessRequest,
  updateAttachmentTag,
  uploadAttachments,
};
