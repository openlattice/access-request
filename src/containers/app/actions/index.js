/*
 * @flow
 */

import { newRequestSequence } from 'redux-reqseq';
import type { RequestSequence } from 'redux-reqseq';

const INITIALIZE_APPLICATION :'INITIALIZE_APPLICATION' = 'INITIALIZE_APPLICATION';
const initializeApplication :RequestSequence = newRequestSequence(INITIALIZE_APPLICATION);

const SET_ROOT :'SET_ROOT' = 'SET_ROOT';
const setRoot = (value :string) => ({
  type: SET_ROOT,
  value
});

export {
  INITIALIZE_APPLICATION,
  initializeApplication,
  SET_ROOT,
  setRoot,
};
