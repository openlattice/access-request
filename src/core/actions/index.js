// @flow

import { newRequestSequence } from 'redux-reqseq';
import type { RequestSequence } from 'redux-reqseq';

const DELETE_BULK_ENTITIES :'DELETE_BULK_ENTITIES' = 'DELETE_BULK_ENTITIES';
const deleteBulkEntities :RequestSequence = newRequestSequence(DELETE_BULK_ENTITIES);

export {
  DELETE_BULK_ENTITIES,
  deleteBulkEntities,
};
