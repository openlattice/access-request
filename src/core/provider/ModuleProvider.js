// @flow

import { ConnectedRouter } from 'connected-react-router/immutable';
import { Provider } from 'react-redux';

import { moduleContext, moduleStore } from '../redux';
import { routerHistory } from '../router';

type Props = {
  children :any;
}

export default function ModuleProvider({ children } :Props) {
  return (
    <Provider context={moduleContext} store={moduleStore}>
      <ConnectedRouter history={routerHistory} context={moduleContext}>
        {children}
      </ConnectedRouter>
    </Provider>
  );
}
