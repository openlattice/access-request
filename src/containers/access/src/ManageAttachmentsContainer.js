// @flow
import { useEffect } from 'react';

// $FlowFixMe
import { List } from 'lattice-ui-kit';
import { DataUtils } from 'lattice-utils';
import type { UUID } from 'lattice';

import AttachmentItem from './AttachmentItem';
import { deleteAttachments, getAttachments } from './actions';
import { ATTACHMENTS } from './reducers/constants';

import { useDispatch, useSelector } from '../../../core/redux';
import { ACCESS } from '../../../core/redux/constants';

const { getEntityKeyId } = DataUtils;
type Props = {
  accessRequestId :UUID;
};

const ManageAttachmentsContainer = ({ accessRequestId } :Props) => {
  const attachments = useSelector((state) => state.getIn([ACCESS, ATTACHMENTS]));
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getAttachments(accessRequestId));
  }, [accessRequestId, dispatch]);

  const onDelete = (fileId) => {
    dispatch(deleteAttachments([fileId]));
  };

  return (
    <List>
      {
        attachments.valueSeq().map((file, index) => {
          const fileId = getEntityKeyId(file);
          const divider = index !== attachments.size - 1;

          return (
            <AttachmentItem
                divider={divider}
                file={file}
                key={fileId}
                onDelete={onDelete} />
          );
        })
      }
    </List>
  );
};

export default ManageAttachmentsContainer;
