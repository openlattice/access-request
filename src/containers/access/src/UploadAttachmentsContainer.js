// @flow
import { Component } from 'react';

import { Map } from 'immutable';
// $FlowFixMe
import { Button, List } from 'lattice-ui-kit';
import { ReduxUtils } from 'lattice-utils';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import type { UUID } from 'lattice';
import type { Dispatch } from 'redux';
import type { RequestSequence } from 'redux-reqseq';

import DocumentItem from './DocumentItem';
import FileUpload from './FileUpload';
import {
  UPLOAD_ATTACHMENTS,
  uploadAttachments
} from './actions';

import { moduleContext } from '../../../core/redux';
import { ACCESS, REQUEST_STATE } from '../../../core/redux/constants';

const { isPending } = ReduxUtils;

type Props = {
  accessRequestId :UUID;
  actions :{
    uploadAttachments :RequestSequence
  },
  loading :boolean;
};

type State = {
  files :Object[];
  tags :string[];
};

class UploadAttachmentsContainer extends Component<Props, State> {

  constructor(props :Props) {
    super(props);
    this.state = this.getInitialState();
  }

  getInitialState = () => ({
    files: [],
    tags: [],
  })

  onDrop = ({ file } :{ file :any }) => {
    const { files } = this.state;
    this.setState({ files: [...files, file] });
  }

  onDelete = (index :number) => {
    const { files } = this.state;
    files.splice(index, 1);
    this.setState({ files });
  };

  onUpload = () => {
    const { accessRequestId, actions } = this.props;
    const { files, tags } = this.state;
    actions.uploadAttachments({
      files,
      tags,
      accessRequestId,
    });
  }

  render() {
    const { files } = this.state;
    const { loading } = this.props;
    return (
      <div>
        <FileUpload onChange={this.onDrop} />
        <List>
          {
            files.map((file, index) => {
              const hasDivider = index !== files.length - 1;
              return (
                <DocumentItem
                    divider={hasDivider}
                    file={file}
                    index={index}
                    onDelete={this.onDelete}
                    onSelectTag={() => {}} />
              );
            })
          }
        </List>
        <Button
            color="primary"
            disabled={!files.length}
            fullWidth
            isLoading={loading}
            onClick={this.onUpload}>
          Upload
        </Button>
      </div>
    );
  }
}

const mapStateToProps = (state :Map) => ({
  loading: isPending(state.getIn([ACCESS, UPLOAD_ATTACHMENTS, REQUEST_STATE]))
});

const mapDispatchToProps = (dispatch :Dispatch<any>) => ({
  actions: bindActionCreators({
    uploadAttachments
  }, dispatch)
});

// $FlowFixMe
export default connect(
  mapStateToProps,
  mapDispatchToProps,
  null,
  { context: moduleContext }
)(UploadAttachmentsContainer);
