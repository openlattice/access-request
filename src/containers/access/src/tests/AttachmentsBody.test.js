import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { Tab, Tabs } from 'lattice-ui-kit';
import { act } from 'react-dom/test-utils';
import { NIL } from 'uuid';

import AttachmentsBody from '../AttachmentsBody';
import ManageAttachmentsContainer from '../ManageAttachmentsContainer';
import ModuleProvider from '../../../../core/provider/ModuleProvider';
import UploadAttachmentsContainer from '../UploadAttachmentsContainer';
import { GET_ATTACHMENTS } from '../actions';

const mockDispatch = jest.fn();
const mockUseDispatch = jest.fn();
mockUseDispatch.mockReturnValue(mockDispatch);

jest.mock('../../../../core/redux', () => {
  const {
    useSelector,
    useStore,
    moduleStore,
    moduleContext,
  } = jest.requireActual('../../../../core/redux');
  return {
    moduleContext,
    moduleStore,
    useDispatch: () => mockUseDispatch(),
    useSelector,
    useStore,
  };
});

describe('AttachmentsBody', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('match snapshot', () => {

    const wrapper = mount(
      <ModuleProvider>
        <AttachmentsBody accessRequestId={NIL} />
      </ModuleProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('show Manage and Upload tabs', () => {
    const wrapper = mount(
      <ModuleProvider>
        <AttachmentsBody accessRequestId={NIL} />
      </ModuleProvider>
    );
    const tabsWrapper = wrapper.find(Tabs);
    const tabWrapper = wrapper.find(Tab);
    expect(tabsWrapper).toHaveLength(1);
    expect(tabWrapper).toHaveLength(2);

    expect(tabWrapper.at(0).prop('label')).toEqual('Manage');
    expect(tabWrapper.at(1).prop('label')).toEqual('Upload');
  });

  test('show ManageAttachmentsContainer by default', () => {
    const wrapper = mount(
      <ModuleProvider>
        <AttachmentsBody accessRequestId={NIL} />
      </ModuleProvider>
    );

    const manageWrapper = wrapper.find(ManageAttachmentsContainer);
    expect(manageWrapper).toHaveLength(1);
    const uploadWrapper = wrapper.find(UploadAttachmentsContainer);
    expect(uploadWrapper).toHaveLength(0);
  });

  test('show UploadAttachmentsContainer on upload tab click', () => {
    const wrapper = mount(
      <ModuleProvider>
        <AttachmentsBody accessRequestId={NIL} />
      </ModuleProvider>
    );

    const tabWrapper = wrapper.find(Tab);
    tabWrapper.at(1).simulate('click');

    const manageWrapper = wrapper.find(ManageAttachmentsContainer);
    expect(manageWrapper).toHaveLength(0);
    const uploadWrapper = wrapper.find(UploadAttachmentsContainer);
    expect(uploadWrapper).toHaveLength(1);
  });

  test('show ManageAttachmentsContainer on upload success', () => {
    const wrapper = mount(
      <ModuleProvider>
        <AttachmentsBody accessRequestId={NIL} />
      </ModuleProvider>
    );

    const tabWrapper = wrapper.find(Tab);
    tabWrapper.at(1).simulate('click');
    const uploadWrapper = wrapper.find(UploadAttachmentsContainer);

    act(() => {
      uploadWrapper.prop('onSuccess')();
    });

    expect(uploadWrapper).toHaveLength(1);
    const manageWrapper = wrapper.find(ManageAttachmentsContainer);
    expect(manageWrapper).toHaveLength(0);
  });

  test('dispatch GET_ATTACHMENTS with accessRequestId', () => {
    const wrapper = mount(
      <ModuleProvider>
        <AttachmentsBody accessRequestId={NIL} />
      </ModuleProvider>
    );

    const attachmentsBodyWrapper = wrapper.find(AttachmentsBody);
    expect(attachmentsBodyWrapper.prop('accessRequestId')).toEqual(NIL);
    expect(mockDispatch).toHaveBeenCalledTimes(1);
    expect(mockDispatch.mock.calls[0][0]).toHaveProperty('id');
    expect(mockDispatch.mock.calls[0][0]).toHaveProperty('type', GET_ATTACHMENTS);
    expect(mockDispatch.mock.calls[0][0]).toHaveProperty('value', NIL);
  });

});
