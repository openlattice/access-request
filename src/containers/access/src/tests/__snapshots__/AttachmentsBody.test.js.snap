// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentsBody match snapshot 1`] = `
<ModuleProvider>
  <Provider
    context={
      Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": Object {},
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      }
    }
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(ConnectedRouterWithContext)
      context={
        Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": Object {},
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        }
      }
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ConnectedRouterWithContext
        context={
          Object {
            "$$typeof": Symbol(react.context),
            "Consumer": Object {
              "$$typeof": Symbol(react.context),
              "_calculateChangedBits": null,
              "_context": [Circular],
            },
            "Provider": Object {
              "$$typeof": Symbol(react.provider),
              "_context": [Circular],
            },
            "_calculateChangedBits": null,
            "_currentRenderer": Object {},
            "_currentRenderer2": null,
            "_currentValue": null,
            "_currentValue2": null,
            "_threadCount": 0,
          }
        }
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        onLocationChanged={[Function]}
      >
        <ConnectedRouter
          context={
            Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": Object {},
              "_currentRenderer2": null,
              "_currentValue": null,
              "_currentValue2": null,
              "_threadCount": 0,
            }
          }
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          onLocationChanged={[Function]}
          store={
            Object {
              "@@observable": [Function],
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        >
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <AttachmentsBody
              accessRequestId="00000000-0000-0000-0000-000000000000"
            >
              <styled__ModalBody>
                <div
                  className="styled__ModalBody-sc-1ycpsik-1 jhOLIL"
                >
                  <ForwardRef
                    indicatorColor="primary"
                    onChange={[Function]}
                    textColor="primary"
                    value={0}
                  >
                    <ForwardRef
                      classes={
                        Object {
                          "centered": "MuiTabs-centered",
                          "fixed": "MuiTabs-fixed",
                          "flexContainer": "MuiTabs-flexContainer",
                          "flexContainerVertical": "MuiTabs-flexContainerVertical",
                          "indicator": "MuiTabs-indicator",
                          "root": "MuiTabs-root",
                          "scrollButtons": "MuiTabs-scrollButtons",
                          "scrollButtonsDesktop": "MuiTabs-scrollButtonsDesktop",
                          "scrollable": "MuiTabs-scrollable",
                          "scroller": "MuiTabs-scroller",
                          "vertical": "MuiTabs-vertical",
                        }
                      }
                      indicatorColor="primary"
                      onChange={[Function]}
                      textColor="primary"
                      value={0}
                    >
                      <div
                        className="MuiTabs-root"
                      >
                        <div
                          className="MuiTabs-scroller MuiTabs-fixed"
                          onScroll={[Function]}
                          style={
                            Object {
                              "marginBottom": null,
                              "overflow": "hidden",
                            }
                          }
                        >
                          <div
                            className="MuiTabs-flexContainer"
                            onKeyDown={[Function]}
                            role="tablist"
                          >
                            <ForwardRef
                              fullWidth={false}
                              indicator={false}
                              key=".0"
                              label="Manage"
                              onChange={[Function]}
                              selected={true}
                              textColor="primary"
                              value={0}
                            >
                              <ForwardRef
                                classes={
                                  Object {
                                    "disabled": "Mui-disabled",
                                    "fullWidth": "MuiTab-fullWidth",
                                    "labelIcon": "MuiTab-labelIcon",
                                    "root": "MuiTab-root",
                                    "selected": "Mui-selected",
                                    "textColorInherit": "MuiTab-textColorInherit",
                                    "textColorPrimary": "MuiTab-textColorPrimary",
                                    "textColorSecondary": "MuiTab-textColorSecondary",
                                    "wrapped": "MuiTab-wrapped",
                                    "wrapper": "MuiTab-wrapper",
                                  }
                                }
                                fullWidth={false}
                                indicator={false}
                                label="Manage"
                                onChange={[Function]}
                                selected={true}
                                textColor="primary"
                                value={0}
                              >
                                <ForwardRef
                                  aria-selected={true}
                                  className="MuiTab-root MuiTab-textColorPrimary Mui-selected"
                                  disabled={false}
                                  focusRipple={true}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  role="tab"
                                  tabIndex={0}
                                >
                                  <ForwardRef
                                    aria-selected={true}
                                    className="MuiTab-root MuiTab-textColorPrimary Mui-selected"
                                    classes={
                                      Object {
                                        "disabled": "Mui-disabled",
                                        "focusVisible": "Mui-focusVisible",
                                        "root": "MuiButtonBase-root",
                                      }
                                    }
                                    disabled={false}
                                    focusRipple={true}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    role="tab"
                                    tabIndex={0}
                                  >
                                    <button
                                      aria-selected={true}
                                      className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
                                      disabled={false}
                                      onBlur={[Function]}
                                      onClick={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      role="tab"
                                      tabIndex={0}
                                      type="button"
                                    >
                                      <span
                                        className="MuiTab-wrapper"
                                      >
                                        Manage
                                      </span>
                                      <ForwardRef
                                        center={false}
                                      >
                                        <ForwardRef
                                          center={false}
                                          classes={
                                            Object {
                                              "child": "MuiTouchRipple-child",
                                              "childLeaving": "MuiTouchRipple-childLeaving",
                                              "childPulsate": "MuiTouchRipple-childPulsate",
                                              "ripple": "MuiTouchRipple-ripple",
                                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                                              "root": "MuiTouchRipple-root",
                                            }
                                          }
                                        >
                                          <span
                                            className="MuiTouchRipple-root"
                                          >
                                            <t
                                              childFactory={[Function]}
                                              component={null}
                                              exit={true}
                                            />
                                          </span>
                                        </ForwardRef>
                                      </ForwardRef>
                                    </button>
                                  </ForwardRef>
                                </ForwardRef>
                              </ForwardRef>
                            </ForwardRef>
                            <ForwardRef
                              fullWidth={false}
                              indicator={false}
                              key=".1"
                              label="Upload"
                              onChange={[Function]}
                              selected={false}
                              textColor="primary"
                              value={1}
                            >
                              <ForwardRef
                                classes={
                                  Object {
                                    "disabled": "Mui-disabled",
                                    "fullWidth": "MuiTab-fullWidth",
                                    "labelIcon": "MuiTab-labelIcon",
                                    "root": "MuiTab-root",
                                    "selected": "Mui-selected",
                                    "textColorInherit": "MuiTab-textColorInherit",
                                    "textColorPrimary": "MuiTab-textColorPrimary",
                                    "textColorSecondary": "MuiTab-textColorSecondary",
                                    "wrapped": "MuiTab-wrapped",
                                    "wrapper": "MuiTab-wrapper",
                                  }
                                }
                                fullWidth={false}
                                indicator={false}
                                label="Upload"
                                onChange={[Function]}
                                selected={false}
                                textColor="primary"
                                value={1}
                              >
                                <ForwardRef
                                  aria-selected={false}
                                  className="MuiTab-root MuiTab-textColorPrimary"
                                  disabled={false}
                                  focusRipple={true}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  role="tab"
                                  tabIndex={-1}
                                >
                                  <ForwardRef
                                    aria-selected={false}
                                    className="MuiTab-root MuiTab-textColorPrimary"
                                    classes={
                                      Object {
                                        "disabled": "Mui-disabled",
                                        "focusVisible": "Mui-focusVisible",
                                        "root": "MuiButtonBase-root",
                                      }
                                    }
                                    disabled={false}
                                    focusRipple={true}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    role="tab"
                                    tabIndex={-1}
                                  >
                                    <button
                                      aria-selected={false}
                                      className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
                                      disabled={false}
                                      onBlur={[Function]}
                                      onClick={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      role="tab"
                                      tabIndex={-1}
                                      type="button"
                                    >
                                      <span
                                        className="MuiTab-wrapper"
                                      >
                                        Upload
                                      </span>
                                      <ForwardRef
                                        center={false}
                                      >
                                        <ForwardRef
                                          center={false}
                                          classes={
                                            Object {
                                              "child": "MuiTouchRipple-child",
                                              "childLeaving": "MuiTouchRipple-childLeaving",
                                              "childPulsate": "MuiTouchRipple-childPulsate",
                                              "ripple": "MuiTouchRipple-ripple",
                                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                                              "root": "MuiTouchRipple-root",
                                            }
                                          }
                                        >
                                          <span
                                            className="MuiTouchRipple-root"
                                          >
                                            <t
                                              childFactory={[Function]}
                                              component={null}
                                              exit={true}
                                            />
                                          </span>
                                        </ForwardRef>
                                      </ForwardRef>
                                    </button>
                                  </ForwardRef>
                                </ForwardRef>
                              </ForwardRef>
                            </ForwardRef>
                          </div>
                          <ForwardRef
                            className="MuiTabs-indicator"
                            color="primary"
                            orientation="horizontal"
                            style={
                              Object {
                                "left": 0,
                                "width": 0,
                              }
                            }
                          >
                            <ForwardRef
                              className="MuiTabs-indicator"
                              classes={
                                Object {
                                  "colorPrimary": "jss2",
                                  "colorSecondary": "jss3",
                                  "root": "jss1",
                                  "vertical": "jss4",
                                }
                              }
                              color="primary"
                              orientation="horizontal"
                              style={
                                Object {
                                  "left": 0,
                                  "width": 0,
                                }
                              }
                            >
                              <span
                                className="jss1 jss2 MuiTabs-indicator"
                                style={
                                  Object {
                                    "left": 0,
                                    "width": 0,
                                  }
                                }
                              />
                            </ForwardRef>
                          </ForwardRef>
                        </div>
                      </div>
                    </ForwardRef>
                  </ForwardRef>
                  <TabPanel
                    index={0}
                    value={0}
                  >
                    <div
                      aria-controls="tabpanel-0"
                      aria-labelledby="tab-0"
                      hidden={false}
                      id="tabpanel-0"
                      role="tabpanel"
                    >
                      <ForwardRef
                        paddingBottom={3}
                        paddingTop={3}
                      >
                        <div
                          className="MuiBox-root jss5"
                        >
                          <ManageAttachmentsContainer
                            accessRequestId="00000000-0000-0000-0000-000000000000"
                          >
                            <ForwardRef>
                              <ForwardRef
                                classes={
                                  Object {
                                    "dense": "MuiList-dense",
                                    "padding": "MuiList-padding",
                                    "root": "MuiList-root",
                                    "subheader": "MuiList-subheader",
                                  }
                                }
                              >
                                <ul
                                  className="MuiList-root MuiList-padding"
                                />
                              </ForwardRef>
                            </ForwardRef>
                          </ManageAttachmentsContainer>
                        </div>
                      </ForwardRef>
                    </div>
                  </TabPanel>
                  <TabPanel
                    index={1}
                    value={0}
                  >
                    <div
                      aria-controls="tabpanel-1"
                      aria-labelledby="tab-1"
                      hidden={true}
                      id="tabpanel-1"
                      role="tabpanel"
                    />
                  </TabPanel>
                </div>
              </styled__ModalBody>
            </AttachmentsBody>
          </Router>
        </ConnectedRouter>
      </ConnectedRouterWithContext>
    </Connect(ConnectedRouterWithContext)>
  </Provider>
</ModuleProvider>
`;
